### PR TITLE
feat(ops): #18 use commitizen

### PR DESCRIPTION
**Pull request by cmp0xff**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,16 @@
 # hamilflow
 
 Dataset of simple physical systems.
+
+## Publish process
+
+- Squash merge all features to `main`
+  - Use `feat(section): xxx`, `fix(section): yyy`, `chore(poetry): lock` etc as the squashed commit messages
+- Run `cz bump --dry` to determine the new version `X.Y.Z`. Or check out the [documentation](https://commitizen-tools.github.io/commitizen/bump/) for imposing a version
+- Run `git checkout release/X.Y.Z` (-b? not sure... I have always been using SourceTree )
```
Comment (emptymalei) [inline README.md:10]: `-b`

File: README.md [hunk 2]
```diff
@@ -1,3 +1,16 @@
...
+- Run `cz bump`. This will
+  - Write `docs/changelog.md` from the commit messages; update the version in `pyproject.toml`
+  - Commit `docs/changelog.md` and `pyproject.toml`
+  - Create a new tag
+- Push to `release/X.Y.Z`
+- Merge to `main`, no need to squash, maybe don't delete the branch under release
```
Comment (emptymalei) [inline README.md:16]: > maybe don't delete the branch under release

This sounds redundant. 
I am quite confused by this. Tags will have exactly the release commit. What is the benefit of clustering the repo with branches. Maybe I missed something?

Comment (cmp0xff) [reply README.md:16]: - `cz` updates `changelog.md` and commits the change
- We then need to push the commit. If we don't create a branch `release/X.Y.Z`, we would have to push the commit directly to `main`.

Comment (emptymalei) [reply README.md:16]: > * `cz` updates `changelog.md` and commits the change
> * We then need to push the commit. If we don't create a branch `release/X.Y.Z`, we would have to push the commit directly to `main`.

Can this stay in the same working branch? For example, this current branch `feature/18-use-commitizen`?

**Pull request merged**